dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/greenplum/GreenplumLogsConnector.java

```diff
@@ -0,0 +1,133 @@
+import com.google.edwmigration.dumper.application.dumper.annotations.RespectsArgumentQueryLogDays;
+import com.google.edwmigration.dumper.application.dumper.annotations.RespectsArgumentQueryLogEnd;
+import com.google.edwmigration.dumper.application.dumper.annotations.RespectsArgumentQueryLogStart;
+import com.google.edwmigration.dumper.application.dumper.annotations.RespectsInput;
+import com.google.edwmigration.dumper.application.dumper.connector.Connector;
+import com.google.edwmigration.dumper.application.dumper.connector.IntervalExpander;
+import com.google.edwmigration.dumper.application.dumper.connector.LogsConnector;
+import com.google.edwmigration.dumper.application.dumper.connector.ZonedInterval;
+import com.google.edwmigration.dumper.application.dumper.connector.ZonedIntervalIterable;
+import com.google.edwmigration.dumper.application.dumper.connector.ZonedIntervalIterableGenerator;
+import com.google.edwmigration.dumper.application.dumper.task.DumpMetadataTask;
+import com.google.edwmigration.dumper.application.dumper.task.JdbcSelectIntervalTask;
+import com.google.edwmigration.dumper.application.dumper.task.JdbcSelectTask;
+import com.google.edwmigration.dumper.application.dumper.task.ParallelTaskGroup;
+import com.google.edwmigration.dumper.application.dumper.task.Task;
+import com.google.edwmigration.dumper.plugin.ext.jdk.annotation.Description;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.GreenplumLogsDumpFormat;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.GreenplumMetadataDumpFormat;
+import java.time.Duration;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@AutoService({Connector.class, LogsConnector.class})
+@Description("Dumps logs from Greenplum")
+@RespectsInput(
+    order = ConnectorArguments.OPT_PORT_ORDER,
+    arg = ConnectorArguments.OPT_PORT,
+    description = "The port of the server.",
+    required = ConnectorArguments.OPT_REQUIRED_IF_NOT_URL,
+    defaultValue = "" + GreenplumLogsConnector.OPT_PORT_DEFAULT
+    )
+@RespectsArgumentAssessment
+@RespectsArgumentQueryLogDays
+@RespectsArgumentQueryLogStart
+@RespectsArgumentQueryLogEnd
+public class GreenplumLogsConnector extends AbstractGreenplumConnector
+    implements LogsConnector, GreenplumLogsDumpFormat {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GreenplumLogsConnector.class);
+
+  public GreenplumLogsConnector() {
+    super("greenplum-logs");
+  }
+
+  @Override
+  public void addTasksTo(List<? super Task<?>> out, ConnectorArguments arguments)
+      throws MetadataDumperUsageException {
+
+    ParallelTaskGroup parallelTask = new ParallelTaskGroup(this.getName());
+    out.add(parallelTask);
+
+    out.add(new DumpMetadataTask(arguments, FORMAT_NAME));
+
+    // is also be there in the metadata , no harm is making zip self-sufficient
```

Comment:
I would suggest removing this comment or rewording it to make it more readable.

Reply:
Done